Client/ContileProvider.swift

```diff
@@ -0,0 +1,67 @@
+//// This Source Code Form is subject to the terms of the Mozilla Public
+//// License, v. 2.0. If a copy of the MPL was not distributed with this
+//// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol ContileProviderInterface {
+    func fetchContiles(completion: @escaping (Result<[Contile], Error>) -> Void)
+}
+
+/// `Contile` is short for contexual tiles. This provider returns "special" metadata and objects for use in Shortcuts (Top Sites) section.
+class ContileProvider: ContileProviderInterface, Loggable {
+    private let contileResourceEndpoint = "https://contile.services.mozilla.com/v1/tiles"
+    
+    public func fetchContiles(completion: @escaping (Result<[Contile], Error>) -> Void) {
+        guard let resourceEndpoint = URL(string: contileResourceEndpoint) else {
+            browserLog.error("Contile Provider - the Contile resource URL is invalid!")
+            completion(.failure(ContileProviderError.invalidResourceEndpoint("Contile Provider - malformed URL.")))
+            return
+        }
+        
+        fetchContilesMeta(resourceEndpoint: resourceEndpoint, completion: completion)
+    }
+    
+    private func fetchContilesMeta(resourceEndpoint: URL, completion: @escaping (Result<[Contile], Error>) -> Void) {
+        let fetchTask = URLSession.shared.dataTask(with: resourceEndpoint) { [weak self] data, response, error in
```

Comment:
`URLSession.shared` -> we have a `makeURLSession` method. This could be made as a lazy var.